### ISSUE TITLE
Moodle 3.10 bugfix

### ISSUE THREAD

**Issue by rabser** [mudrd8mz/moodle-report_overviewstats#10]
Using this plugin under Moodle 3.10 dumps an error about the coursecatlib not available anymore.

You can get it running again, simply modifying the file classes/chart_courses.php as follow:

comment out the require_once 
```
//require_once($CFG->libdir.'/coursecatlib.php');
```
and modify all the coursecat:: instances with core_course_category::

```
        //$cats = coursecat::make_categories_list();
        $cats = core_course_category::make_categories_list();
        $this->data['percategory'] = array();
        $total = 0;

        foreach ($cats as $catid => $catname) {
            //$cat = coursecat::get($catid);
            $cat = core_course_category::get($catid);
....

**Comment by sei-ebram** [comment]
Refer to Moodle 3.9 documentation where it warns that this library should no longer be references, how how to mitigate:

https://github.com/moodle/moodle/blob/MOODLE_39_STABLE/lib/coursecatlib.php#L28-L31

**Comment by sei-ebram** [comment]
@mudrd8mz

**Comment by sei-ebram** [comment]
Scrolling back through the Moodle versions, it looks like this fix will be valid in Moodle 3.6+

Moodle 3.6 mentioning the same thing:
https://github.com/moodle/moodle/blob/MOODLE_36_STABLE/lib/coursecatlib.php#L28-L31

**Comment by mudrd8mz** [comment]
Thank you @sei-ebram for the fix. It is now part of version 1.4 of the plugin.
